### ISSUE TITLE
Jenkins html report not displaying nicely

### ISSUE THREAD

**Issue by petrheckoHims** [DannyDainton/newman-reporter-htmlextra#45]
Hello,

Firstly, thanks a lot for this amazing html reporter, it looks great! I was wondering if there is any way to make it work with jenkins html publisher (or if I'm doing something wrong on my end?). If I run newman locally with the exact same parameters, I get a nice report as expected. If I run the same in Jenkins via this command:

```
newman run "https://api.getpostman.com/collections/${uid}?apikey=${postman_api}" \
--reporters cli,junit,htmlextra \
--reporter-junit-export "newman/report.xml" \
--reporter-htmlextra-export "newman/report.html"
```
It will generate the report.html file just fine. The problem is that when using the jenkins plugin, `HTML Publisher`, the report is displayed all over the place, making it not really readable. I also have an email notification set which will email me the report.html file, however, it's the same issue when clicking on the report from the email will not display it properly, but I need to actually download the report from the email first and then open it - this way the report will display correctly, but it's more steps to achieve it.

Any advice will be appreciated.

**Comment by DannyDainton** [comment]
Hey,

Thanks for the feedback. 

I don't really know anything about that plugin or know what it's doing in Jenkins. 

If the information is there but the styling isn't, it could be blocking some of the CSS/JS references in the file. Have you checked the Dev Tools for any errors.

Sorry, I really have no idea what's going on with that as it's not an area that I have experience of.

**Comment by petrheckoHims** [comment]
Thanks Danny, I will try to troubleshoot some more - seems like it has to do with Content Security Policy as when disabled in Jenkins, the report will display (but that's not safe), so I will keep troubleshooting.

**Comment by DannyDainton** [comment]
OK, I'm doing to close this for now. We can reopen if you find a reporter specific issue.

Interesting to know the outcome of your investigation though.

**Comment by bmakamsatyaprakash** [comment]
This is how the report is looking like on Jenkins pipeline 
<img width="748" alt="Screen Shot 2019-06-20 at 3 54 52 pm" src="https://user-images.githubusercontent.com/36174879/59824215-ceb4fa80-9373-11e9-806c-7d512eb95a16.png">

**Comment by bmakamsatyaprakash** [comment]
and looking for that to look like this 
<img width="1234" alt="Screen Shot 2019-06-20 at 3 56 11 pm" src="https://user-images.githubusercontent.com/36174879/59824271-fe640280-9373-11e9-9f8c-882bdecb6ad5.png">

**Comment by DannyDainton** [comment]
That's doesn't really have anything to do with the reporter - It's a Jenkins confirmation thing that I have zero control over.

This was a solution that worked for someone else over in the Postman community. 

https://community.getpostman.com/t/newman-htmlextra-reporter-not-displaying-correctly-when-viewed-from-jenkins/5735/2

**Comment by petrheckoHims** [comment]
I'm able to see the nice version of the newman report after changing the jenkins CSP to:

`sandbox allow-same-origin allow-scripts; default-src 'self'; script-src 'unsafe-eval' *; img-src *; style-src 'unsafe-inline' *; font-src *;`

You can refer to [this](https://www.cyotek.com/blog/adjusting-the-jenkins-content-security-policy) article for more details of how to make the CSP change.

**Comment by bmakamsatyaprakash** [comment]
Hi ,

Really appreciate your help.. It worked  :) !!!

Thank You very much for helping me and making it would in no time. This was very crucial for me and you helped me to get the issue resolved .
Running this on terminal as per your details worked perfectly fine giving me nice reports.

java -Dhudson.model.DirectoryBrowserSupport.CSP="sandbox allow-same-origin allow-scripts; default-src 'self'; script-src 'unsafe-eval' *; img-src *; style-src 'unsafe-inline' *; font-src *;" -jar jenkins.war --httpPort=8082

Regards,
Bindhu Makam
Senior Consultant | Platform Engineering
Deloitte Consulting Pty Ltd
225 George Street, Sydney, NSW, 2000, Australia
Tel : Mobile: +61 403146677
bmakamsatyparakash@deloitte.com.au<mailto:bmakamsatyparakash@deloitte.com.au> | www.deloitte.com.au<http://www2.deloitte.com/au/en.html?utm_source=outlook&utm_medium=email&utm_campaign=exc-outlook-signature&utm_content=text>

**Comment by jimmymccauleybp** [comment]
Hi @bmakamsatyaprakash @petrheckoHims 
Same issue has been causing me a few problems too! I have been wasting my time looking at NPM dependencies which is not the correct solution. I'm glad it's been resolved for you. 
Could I please ask where exactly you run this command? I am assuming you will need a Jenkins administrator to execute this command, if you don't have admin?
Appreciate you confirmation; Thanks!

**Comment by bmakamsatyaprakash** [comment]
Hi ,

I ran below command on terminal (mac) when starting up my jenkins. Jenkins is on my local machine for now.
After running below command, I got nice reports. Please try and let me know.


“java -Dhudson.model.DirectoryBrowserSupport.CSP="sandbox allow-same-origin allow-scripts; default-src 'self'; script-src 'unsafe-eval' *; img-src *; style-src 'unsafe-inline' *; font-src *;" -jar jenkins.war --httpPort=8082”


Regards,
Bindhu Makam
Senior Consultant | Platform Engineering
Deloitte Consulting Pty Ltd
225 George Street, Sydney, NSW, 2000, Australia
Tel : Mobile: +61 403146677
bmakamsatyparakash@deloitte.com.au<mailto:bmakamsatyparakash@deloitte.com.au> | www.deloitte.com.au<http://www2.deloitte.com/au/en.html?utm_source=outlook&utm_medium=email&utm_campaign=exc-outlook-signature&utm_content=text>

**Comment by jimmymccauleybp** [comment]
Hi @bmakamsatyaprakash 
Thanks for replying so quickly. 
I've passed the info onto the Jenkins admin who executed the command. It's all working nicely now. 
All the best!

**Comment by DannyDainton** [comment]
Good to see you got it sorted! 

Please feel free to open new issues, if you find something specifically wrong with the report itself.

**Comment by jimmymccauleybp** [comment]
Cheers Danny - and kudos! the enhanced report is awesome....

**Comment by bmakamsatyaprakash** [comment]
That’s wonderful!
Thanks for letting me know.

Regards,
Bindhu Makam
Senior Consultant | Platform Engineering
Deloitte Consulting Pty Ltd
225 George Street, Sydney, NSW, 2000, Australia
Tel : Mobile: +61 403146677
bmakamsatyparakash@deloitte.com.au<mailto:bmakamsatyparakash@deloitte.com.au> | www.deloitte.com.au<http://www2.deloitte.com/au/en.html?utm_source=outlook&utm_medium=email&utm_campaign=exc-outlook-signature&utm_content=text>

**Comment by bmakamsatyaprakash** [comment]
Yes.. Someone from your team replied and helped me to get this resolved and helped a lot in my demo. Cheers!
Sure, thanks for all the support.


Regards,
Bindhu Makam
Senior Consultant | Platform Engineering
Deloitte Consulting Pty Ltd
225 George Street, Sydney, NSW, 2000, Australia
Tel : Mobile: +61 403146677
bmakamsatyparakash@deloitte.com.au<mailto:bmakamsatyparakash@deloitte.com.au> | www.deloitte.com.au<http://www2.deloitte.com/au/en.html?utm_source=outlook&utm_medium=email&utm_campaign=exc-outlook-signature&utm_content=text>

**Comment by Dohbedoh** [comment]
A better solution than using the CSP header / relaxing CSP rules in Jenkins is to setup a **Resources Root URL**, available since Jenkins 2.200: https://www.jenkins.io/doc/upgrade-guide/2.204/#resource-domain-support
